src/tests/test_parse.py

```diff
@@ -298,28 +299,73 @@ def complete_with_defaults(mapping):
 
 
 @pytest.mark.parametrize(
-    "line, expected",
+    "line, cmd_from, expected_error_message",
     [
         (
-            "some-argument",
-            {"line": ["some-argument"]},
+            "duckdb:// --alias test1 --alias test2",
+            "sql",
+            (
+                "Duplicate arguments in %sql. "
+                "Please use only one of each of the following: --alias"
+            ),
         ),
         (
-            "a b c",
-            {"line": ["a", "b", "c"]},
+            """histogram --table penguins.csv --column bill_length_mm
+--column body_mass_g""",
+            "sqlplot",
+            (
+                "Duplicate arguments in %sqlplot. "
+                "Please use only one of each of the following: --column"
+            ),
         ),
+        (
+            """bar --table penguins.csv --column bill_length_mm
+--show-numbers --show-numbers""",
+            "sqlplot",
+            (
+                "Duplicate arguments in %sqlplot. "
+                "Please use only one of each of the following: --show-numbers"
+            ),
+        ),
+    ],
+)
+def test_magic_args_raises_usageerror(
```

Comment:
add some tests that use short arguments (single leading dash)

Reply:
I assigned this test to test the entire logic by running the actual sql magic commands. It now has all the tests (for double hyphen, single hyphen and mixed).